File: src/main/java/com/athar/bms/product/service/ProductService.java
package com.athar.bms.product.service;

import com.athar.bms.product.dto.ProductRequest;
import com.athar.bms.product.dto.ProductResponse;

import java.util.List;

public interface ProductService {

    ProductResponse createProduct(ProductRequest request);

    List<ProductResponse> getAllProducts();

    ProductResponse getProductById(Long id);

    ProductResponse updateProduct(Long id, ProductRequest request);

    void deleteProduct(Long id);
}

File: frontend/src/services/productService.js
import api from "./api";

export const getAllProducts = async () => {
    const response = await api.get("/products");
    return response.data;
};

export const createProduct = async (product) => (await api.post("/products", product)).data;
export const updateProduct = async (id, product) => (await api.put(`/products/${id}`, product)).data;
export const deleteProduct = async (id) => api.delete(`/products/${id}`);